src/main/java/hudson/plugins/checkstyle/CheckStyleResult.java

```diff
@@ -71,6 +128,15 @@ protected CheckStyleResult(final AbstractBuild<?, ?> build, final String default
         }
     }
 
+    CheckStyleResult(final Run<?, ?> build, final BuildHistory history,
+                     final ParserResult result, final String defaultEncoding, final boolean canSerialize) {
+        super(build, history, result, defaultEncoding);
+
+        if (canSerialize) {
+            serializeAnnotations(result.getAnnotations());
+        }

```

Comment:
I don't see how this `canSerialize` was used before. It was in the old signatures, but doesn't look (from the diff anyway) as though it was used in this way. Just saying in case this is a new behaviour that might cause a regression.